src/consumer/runner.js

```diff
@@ -416,6 +434,16 @@ module.exports = class Runner extends EventEmitter {
           })
           return
         }
+        if (this.consumerGroup.isPaused(batch.topic, batch.partition)) {
```

Comment:
I wonder if this needs to come after the `if (isRebalancing(e) ...` condition, since I would assume that even if this topic-partition is paused, we would still want to rejoin the group and potentially receive new assignments.

Reply:
Yes, that sounds right to me.